qiskit/aqua/algorithms/distribution_learners/qgan.py

```diff
@@ -278,6 +281,9 @@ def train(self):
                 writer = csv.DictWriter(csv_file, fieldnames=fieldnames)
                 writer.writeheader()
 
+        if len(self._data) < self._batch_size:
+            raise AquaError('Please reduce the batch size.')
```

Comment:
Could we make the error message include a bit more guidance? E.g.
`'The batch size needs to be less than the truncated data size of {}.format(len(self._data))'` otherwise the message does not really help me much on how I should reduce it and maybe the user just ends up doing it by trial and error otherwise